numpy/lib/histograms.py

```diff
@@ -163,12 +163,20 @@ def _hist_bin_fd(x):
 def _hist_bin_auto(x):
     """
     Histogram bin estimator that uses the minimum width of the
-    Freedman-Diaconis and Sturges estimators.
+    Freedman-Diaconis and Sturges estimators if the FD bandwidth is non zero
+    and the Sturges estimator if the FD bandwidth is 0.
 
     The FD estimator is usually the most robust method, but its width
-    estimate tends to be too large for small `x`. The Sturges estimator
-    is quite good for small (<1000) datasets and is the default in the R
-    language. This method gives good off the shelf behaviour.
+    estimate tends to be too large for small `x` and bad for data with limited
+    variance. The Sturges estimator is quite good for small (<1000) datasets
+    and is the default in the R language. This method gives good off the shelf
+    behaviour.
+
+    .. versionchanged:: 1.15.0
+    If there is limited variance, the IQR will be 0 which will result in
+    1 bin being chosen which may not be optimal. If the IQR is 0, it's
```

Comment:
Perhaps this should say that "when IQR is 0, the bin width from the FD estimator is 0, so instead ...". "1 bin being chosen" is arguably a bug in `histogram_bin_edges`

Reply:
To be clear, `1` is chosen not due to any hope that it is optimal, but because it is a more useful choice than `np.inf`.

Reply:
when no data is passed in or the data has no variance, 1 bin is correct, just not always.